### PR TITLE
Use more opcache & supress PHP warnings

### PR DESCRIPTION
**Pull request by RogoTe**
- install opcache-modul (should come with php7.0 by default, but to be shure)
- create config file
opcache should be reused even after restart of raspi
goto http://phoniebox.local/php/opcache to see status
https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/554

- supress PHP warnings and notices https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/378

### CODE REVIEW

File: scripts/installscripts/stretch-install-default.sh [hunk 1]
```diff
@@ -387,7 +387,7 @@ sudo iwconfig wlan0 power off
 
 # Install required packages
 sudo apt-get update
-sudo apt-get --yes --force-yes install apt-transport-https samba samba-common-bin python-dev python-pip gcc linux-headers-4.9 lighttpd php7.0-common php7.0-cgi php7.0 php7.0-fpm at mpd mpc mpg123 git ffmpeg python-mutagen python3-gpiozero
+sudo apt-get --yes --force-yes install apt-transport-https samba samba-common-bin python-dev python-pip gcc linux-headers-4.9 lighttpd php7.0-common php7.0-cgi php7.0 php7.0-fpm install php7.0-mbstring php7.0-opcache at mpd mpc mpg123 git ffmpeg python-mutagen python3-gpiozero
```
Comment (ZyanKLee) [inline scripts/installscripts/stretch-install-default.sh]: there seems to be an "install" too many in there
```suggestion
sudo apt-get --yes --force-yes install apt-transport-https samba samba-common-bin python-dev python-pip gcc linux-headers-4.9 lighttpd php7.0-common php7.0-cgi php7.0 php7.0-fpm php7.0-mbstring php7.0-opcache at mpd mpc mpg123 git ffmpeg python-mutagen python3-gpiozero
```

File: README.md [hunk 2]
```diff
@@ -27,8 +27,9 @@ Celebrating all the great designs of 2018, I put together a calendar for 2019. T
 ---
 
 <a href="https://www.buymeacoffee.com/MiczFlor" target="_blank"><img src="https://www.buymeacoffee.com/assets/img/guidelines/download-assets-sm-2.svg"></a>
+<img src="https://web4africa.com/wp-content/uploads/paypal.png" width="30px" height="30px">
```
Comment (s-martin) [inline README.md]: Why is there a link to web4africa.com with a Paypal image?

Comment (ZyanKLee) [inline README.md]:
```suggestion
```

**Comment by ZyanKLee** [comment]
@RogoTe please resolve those conflicts by either rebasing with MiczFlor's develop branch or by merging that same branch's changes into this one.

File: scripts/installscripts/stretch-install-default.sh [hunk 3]
```diff
@@ -440,6 +440,20 @@ sudo cp /home/pi/RPi-Jukebox-RFID/misc/sampleconfigs/php.ini.stretch-default.sam
 sudo chown root:root /etc/php/7.0/fpm/php.ini
 sudo chmod 644 /etc/php/7.0/fpm/php.ini
 
+# settings for opcache
+# we cache to memory and disk (level 2) for faster upstart
+sudo mkdir /home/pi/RPi-Jukebox-RFID/.php-opcache/
+sudo chmod 1777 /home/pi/RPi-Jukebox-RFID/.php-opcache/
+sudo touch /etc/php/7.0/fpm/conf.d/999-RPi-Jukebox-RFID.ini
```
Comment (s-martin) [inline scripts/installscripts/stretch-install-default.sh]: fpm was removed now in #839, should this be removed also here?

File: scripts/installscripts/stretch-install-default.sh [hunk 4]
```diff
@@ -387,7 +387,7 @@ sudo iwconfig wlan0 power off
 
 # Install required packages
 sudo apt-get update
-sudo apt-get --yes --force-yes install apt-transport-https samba samba-common-bin python-dev python-pip gcc linux-headers-4.9 lighttpd php7.0-common php7.0-cgi php7.0 php7.0-fpm at mpd mpc mpg123 git ffmpeg python-mutagen python3-gpiozero
+sudo apt-get --yes --force-yes install apt-transport-https samba samba-common-bin python-dev python-pip gcc linux-headers-4.9 lighttpd php7.0-common php7.0-cgi php7.0 php7.0-fpm php7.0-mbstring php7.0-opcache at mpd mpc mpg123 git ffmpeg python-mutagen python3-gpiozero
```
Comment (s-martin) [inline scripts/installscripts/stretch-install-default.sh]: fpm was removed in #839, don't introduce here again?

File: scripts/installscripts/stretch-install-spotify.sh [hunk 5]
```diff
@@ -456,7 +456,7 @@ sudo iwconfig wlan0 power off
 
 # Install required packages
 sudo apt-get update
-sudo apt-get --yes --force-yes install apt-transport-https samba samba-common-bin python-dev python-pip gcc linux-headers-4.9 lighttpd php7.0-common php7.0-cgi php7.0 php7.0-fpm at mpd mpc mpg123 git ffmpeg python-mutagen python3-gpiozero
+sudo apt-get --yes --force-yes install apt-transport-https samba samba-common-bin python-dev python-pip gcc linux-headers-4.9 lighttpd php7.0-common php7.0-cgi php7.0 php7.0-fpm php7.0-mbstring php7.0-opcache at mpd mpc mpg123 git ffmpeg python-mutagen python3-gpiozero
```
Comment (s-martin) [inline scripts/installscripts/stretch-install-spotify.sh]: see above

File: scripts/installscripts/stretch-install-spotify.sh [hunk 6]
```diff
@@ -531,6 +531,20 @@ sudo cp /home/pi/RPi-Jukebox-RFID/misc/sampleconfigs/php.ini.stretch-default.sam
 sudo chown root:root /etc/php/7.0/fpm/php.ini
 sudo chmod 644 /etc/php/7.0/fpm/php.ini
 
+# settings for opcache
+# we cache to memory and disk (level 2) for faster upstart
+sudo mkdir /home/pi/RPi-Jukebox-RFID/.php-opcache/
+sudo chmod 1777 /home/pi/RPi-Jukebox-RFID/.php-opcache/
+sudo touch /etc/php/7.0/fpm/conf.d/999-RPi-Jukebox-RFID.ini
```
Comment (s-martin) [inline scripts/installscripts/stretch-install-spotify.sh]: see above

File: scripts/installscripts/stretch-install-default.sh (hunk 4 above)
Comment (ZyanKLee) [reply scripts/installscripts/stretch-install-default.sh]: nah, that would not be a wise move :-D
That's why there are conflicts with the develop branch.

File: scripts/installscripts/stretch-install-default.sh (hunk 3 above)
Comment (ZyanKLee) [reply scripts/installscripts/stretch-install-default.sh]: though these configs would be better of in the `/etc/php/7.0/cgi` directory

**Comment by s-martin** [comment]
@RogoTe, if you resolve the conflicts this PR could be merged.

**Comment by s-martin** [comment]
How should we proceed on this PR?

**Comment by s-martin** [comment]
PR has been dead since quite a while now and stretch isn’t supported anymore, so I’m closing this now. Feel free to reopen it.

**Pull request closed without merging**